### PR TITLE
unit test s3dis_1x1

### LINKED ISSUE #2: Add Pytorch Geometric S3DIS Dataset for Semantic Segmentation Testing
https://pytorch-geometric.readthedocs.io/en/latest/modules/datasets.html#torch_geometric.datasets.S3DIS

### PR DESCRIPTION
**Pull request by Stakhan**
## What does this PR do?
+ Add unit test for `src.datamodules.segmentation.s3dis_1x1_module`
+ Fix unit test for `src.datamodules.classification.modelnet2048_module`

Fixes #2

## Before submitting

- [ ] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **test your PR locally** with `pytest` command?
- [ ] Did you **run pre-commit hooks** with `pre-commit run -a` command?

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,6 +3,7 @@ torch>=1.10.0
 torchvision>=0.11.0
 pytorch-lightning==1.7.1
 torchmetrics==0.10.0
+torch_geometric>=2.1.0
```
Comment (CCInc) [inline requirements.txt:6]: I think this is the wrong package for pip, I'm showing it as "torch-geometric". We will probably also need to add "torch-scatter" and "torch-sparse" at minimum to get the tests running in the CI environment

File: tests/test_s3dis_1x1.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    dm = S3DIS1x1DataModule(config=config)
+    dm.prepare_data()
+    assert not dm.data_train and not dm.data_test
+
+    expected_data_location = Path(config.data_dir)
+    assert expected_data_location.exists()
+
+    dm.setup()
+    assert dm.data_train and dm.data_test
+    assert dm.train_dataloader() and dm.test_dataloader()
```
Comment (CCInc) [inline tests/test_s3dis_1x1.py:25]: we can probably check the length of the datasets as well like we do for modelnet

File: tests/test_s3dis_1x1.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    batch = next(iter(dm.train_dataloader()))
+    x, y = batch.pos, batch.y
+    print(x.shape, y.shape)
+    assert len(x) == batch_size
+    assert len(y) == batch_size
+
+    # rm created folder
+    shutil.rmtree(expected_data_location)
```
Comment (CCInc) [inline tests/test_s3dis_1x1.py:34]: maybe for s3dis and modelnet we can add tests that validate the num_classes and num_feats field that were recently added?

File: tests/test_modelnet2048.py [hunk 4]
```diff
@@ -3,18 +3,23 @@
...
-    dm = ModelNet2048DataModule(data_dir=data_dir, batch_size=batch_size)
+    dm = ModelNet2048DataModule(config=config)
     dm.prepare_data()
 
     assert not dm.data_train and not dm.data_test
-    assert Path(data_dir).exists()
+    assert Path(config.data_dir).exists()
 
     dm.setup()
     assert dm.data_train and dm.data_test
```
Comment (CCInc) [inline tests/test_modelnet2048.py:25]: For this one we can also validate that batch.x is empty/null

File: tests/test_s3dis_1x1.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    expected_data_location = Path(config.data_dir)
+    assert expected_data_location.exists()
+
+    dm.setup()
+    assert dm.data_train and dm.data_test
+    assert dm.train_dataloader() and dm.test_dataloader()
+
+    batch = next(iter(dm.train_dataloader()))
+    x, y = batch.pos, batch.y
```
Comment (CCInc) [inline tests/test_s3dis_1x1.py:28]: maybe we can check that there are 6 feature channels (`batch.x`), 3 pos channels `batch.pos` and validate the shape of batch.y?

**Review by CCInc — COMMENTED** [review]
Thanks!

File: tests/test_s3dis_1x1.py (hunk 5 above)
Comment (leo-stan) [reply tests/test_s3dis_1x1.py:28]: Addressed in #30

File: tests/test_modelnet2048.py (hunk 4 above)
Comment (leo-stan) [reply tests/test_modelnet2048.py:25]: Addressed in #30

File: tests/test_s3dis_1x1.py (hunk 2 above)
Comment (leo-stan) [reply tests/test_s3dis_1x1.py:25]: Addressed in #30

**Comment by CCInc** [comment]
Fully integrated into #30 , feel free to branch off there with further changes!

**Pull request closed without merging**